src/ctapipe/image/muon/features.py

```diff
@@ -158,3 +158,113 @@ def ring_containment(radius, center_x, center_y, camera_radius):
+    image,
+    image_mask,
+):
+    """
+    Calculate the parameters related to the size of the ring image.
+
+    Parameters
+    ----------
+    radius: float
+        radius of the ring
+    center_x: float
+        x coordinate of the ring center
+    center_y: float
+        y coordinate of the ring center
+    pixel_x: array-like
+        x coordinates of the camera pixels
+    pixel_y: array-like
+        y coordinates of the camera pixels
+    ring_integration_width: float
+        Width of the ring in fractions of ring radius
+    outer_ring_width: float
+        Width of the outer ring in fractions of ring radius
+    image: array-like
+        Amplitude of image pixels
+    image_mask: array-like
+        mask of the camera pixels after cleaning
+
+    Returns
+    -------
+    ring_size: float
+        Sum of the p.e. inside the integration area of the ring
+    size_outside: float
+        Sum of the photons outside the ring integration area that passed the cleaning
+    num_pixels_in_ring: int
+        Number of pixels inside the ring integration area that passed the cleaning
+    mean_pixel_outside_ring: float
+        Mean intensity of the pixels outside the ring, but still close to it
+    """
+
+    dist = np.sqrt((pixel_x - center_x) ** 2 + (pixel_y - center_y) ** 2)
+    dist_mask = np.abs(dist - radius) < (radius * ring_integration_width)
+    pix_ring = image * dist_mask
+    pix_outside_ring = image * ~dist_mask
+
+    dist_mask_2 = np.logical_and(
+        ~dist_mask,
+        np.abs(dist - radius) < radius * (ring_integration_width + outer_ring_width),
+    )
+    pix_ring_2 = image[dist_mask_2]
+
+    ring_size = np.sum(pix_ring)
+    size_outside = np.sum(pix_outside_ring * image_mask)
+    num_pixels_in_ring = np.sum(dist_mask & image_mask)
+    mean_pixel_outside_ring = np.sum(pix_ring_2) / len(pix_ring_2)
+
+    return ring_size, size_outside, num_pixels_in_ring, mean_pixel_outside_ring
+
+
+def radial_light_distribution(center_x, center_y, pixel_x, pixel_y, image):
+    """
+    Calculate the radial distribution of the muon ring.
+
+    Parameters
+    ----------
+    center_x : float
+        x coordinate of the ring center.
+    center_y : float
+        y coordinate of the ring center.
+    pixel_x : array-like
+        x coordinates of the camera pixels.
+    pixel_y : array-like
+        y coordinates of the camera pixels.
+        Amplitude of image pixels.
+
+    Returns
+    -------
+    standard_dev : float
+        Standard deviation of the light distribution along the ring radius.
+    skewness : float
+        Skewness of the light distribution along the ring radius.
+    excess_kurtosis : float
+        Excess kurtosis of the light distribution along the ring radius.
+    """
+
+    if np.sum(image) == 0:
+        return np.nan * u.deg, np.nan, np.nan
+
+    x0 = center_x.to_value(u.deg)
+    y0 = center_y.to_value(u.deg)
+    pix_x = pixel_x.to_value(u.deg)
+    pix_y = pixel_y.to_value(u.deg)
```

Comment:
I don't think you need to convert and take the value here. Numpy works fine with unit quantities, and then the function will work no matter what unit the inputs are in.